src/core/server/uuid/uuid_service.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { manageInstanceUuid } from './manage_uuid';
+import { CoreContext } from '../core_context';
+import { Logger } from '../logging';
+import { IConfigService } from '../config';
+import { LegacyServiceDiscoverPlugins } from '../legacy/legacy_service';
+
+/**
+ * APIs to access the application's instance uuid.
+ */
+export interface UuidServiceSetup {
+  /**
+   * Retrieve the Kibana instance uuid.
+   */
+  getInstanceUuid(): string;
+}
+
+interface SetupDeps {
+  legacyPlugins: LegacyServiceDiscoverPlugins;
+}
+
+/** @internal */
+export class UuidService {
+  private readonly log: Logger;
+  private readonly configService: IConfigService;
+  private uuid: string = '';
+
+  constructor(core: CoreContext) {
+    this.log = core.logger.get('uuid');
+    this.configService = core.configService;
+  }
+
+  public async setup({ legacyPlugins }: SetupDeps) {
+    this.uuid = await manageInstanceUuid(this.configService);
+    this.log.info(`Kibana instance UUID: ${this.uuid}`);
```

Comment:
The legacy was logging different (but very similar) messages depending on where the uuid was read from. I think we don't really care, and that the only important info is the actual UUID, so I simplified. Please tell me if I'm wrong.

Reply:
I'm not sure if we necessarily care, but seeing how this does affect startup and some critical functions (task_manager) maybe it would be good to retain some insight. That said, I think downgrading the detailed messages to debug would be fine.

Reply:
Ok, adapted to use the exact same messages we had in legacy, and used debug instead of info